### PR TITLE
Finish app

### PR DESCRIPTION
**Pull request by m-talbi**
- Added Sign up and log in pages.
- Added categories page
- Added expenses page
- Added ability to add new categories and expenses and delete them.
- Added authentication and authorization.
- Added model and integration tests.

Presentation: [video](https://streamable.com/pjlufd)

### CODE REVIEW

File: app/helpers/expenses_helper.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+module ExpensesHelper
```
Comment (arslanbisharat) [inline app/helpers/expenses_helper.rb:1]: - [ ] When I try to save a category without entering a value, to would show this: 
image

Let's fix this for the app to work properly.

File: app/helpers/expenses_helper.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+  def sum_amounts(expenses)
```
Comment (arslanbisharat) [inline app/helpers/expenses_helper.rb:2]: - [ ] If an input filed is empty, it gives the error; 
image

File: test/channels/application_cable/connection_test.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+require 'test_helper'
```
Comment (arslanbisharat) [inline test/channels/application_cable/connection_test.rb:1]: - [ ] Let's make sure you have required tests for all the important functions as requested in project
image

**Review by arslanbisharat — CHANGES_REQUESTED** [review]
Hi @Kweeka1, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

- The Pull Request has a proper title and description. ✔️
- You have added a descriptive Readme file. Good Job. ✔️
- There is no linter error is present. ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- [ ] Please make sure you right descriptive comments. Commenting involves placing Human Readable Descriptions inside of computer programs detailing what the Code is doing. Proper use of commenting can make code maintenance much easier, as well as helping make finding bugs faster. Further, commenting is very important when writing functions that other people will use. :x:


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: test/channels/application_cable/connection_test.rb (hunk 3 above)
Comment (VICTOR-CHUKWU) [reply test/channels/application_cable/connection_test.rb:1]: - [ ] issues well resolved :+1: 
image

**Review by VICTOR-CHUKWU — APPROVED** [review]
Hi @Kweeka1 ,
# STATUS: APPROVED! 🥇 💯 
done

In my opinion, your project is ready for the presentation!  There is nothing else to say other than... it's time to merge it :shipit: 


Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**